src/middlewared/middlewared/plugins/virt/instance_device.py

```diff
@@ -68,6 +68,11 @@ async def incus_to_device(self, name: str, incus: dict[str, Any], context: dict)
             case 'nic':
                 device['dev_type'] = 'NIC'
                 device['network'] = incus.get('network')
+                if device['network']:
+                    device['parent'] = None
+                else:
+                    device['nic_type'] = incus.get('nictype').upper()
```

Comment:
This will crash with AttributeError if `nictype` doesn't exist.

Reply:
Good catch, fixed